driver/compute_server_client.py

```diff
@@ -108,9 +122,34 @@ def post_table_level_observation(self, data: TableLevelObservation) -> None:
         headers["Content-Type"] = "application/json"
         try:
             response = self._req_session.post(
-                url, data=data_str, headers=headers, timeout=10
+                url, data=data_str, headers=headers, timeout=TIMEOUT_SEC
             )
             response.raise_for_status()
         except Exception as ex:
             msg = "Failed to post the table level observation to the server"
             raise ComputeServerClientException(msg, ex) from ex
+
+    def post_query_observation(self, data: QueryObservation) -> None:
+        """Post **COMPRESSED** query observation to the server
+        Args:
+            session_id: Session that the data is uploaded to.
+            data: Collected data from the target database.
+        Raises:
+            ComputeServerClientException: Failed to post the observation.
+        """
+        url = f"{self._server_url}/query_observation/"
+        # pylint: disable=c-extension-no-member
+        compressed_data = zlib.compress(json.dumps(data, indent=2).encode('utf-8'))
```

Comment:
wondering will indent=2 make the data a bit larger 😂

Reply:
Yeah but I think this is okay with compression.